packages/editor-ui/src/stores/ui.store.ts

```diff
@@ -87,545 +81,660 @@ try {
+					title: 'contextual.workflows.sharing.title',
+					unavailable: {
+						title: `contextual.workflows.sharing.unavailable.title${contextKey}`,
+						description: {
+							modal: `contextual.workflows.sharing.unavailable.description.modal${contextKey}`,
+							tooltip: `contextual.workflows.sharing.unavailable.description.tooltip${contextKey}`,
 						},
+						action: `contextual.workflows.sharing.unavailable.action${contextKey}`,
+						button: `contextual.workflows.sharing.unavailable.button${contextKey}`,
 					},
 				},
-				workflows: {
-					sharing: {
-						title: 'contextual.workflows.sharing.title',
-						unavailable: {
-							title: `contextual.workflows.sharing.unavailable.title${contextKey}`,
-							description: {
-								modal: `contextual.workflows.sharing.unavailable.description.modal${contextKey}`,
-								tooltip: `contextual.workflows.sharing.unavailable.description.tooltip${contextKey}`,
-							},
-							action: `contextual.workflows.sharing.unavailable.action${contextKey}`,
-							button: `contextual.workflows.sharing.unavailable.button${contextKey}`,
-						},
-					},
+			},
+			variables: {
+				unavailable: {
+					title: `contextual.variables.unavailable.title${contextKey}`,
+					description: 'contextual.variables.unavailable.description',
+					action: `contextual.variables.unavailable.action${contextKey}`,
+					button: `contextual.variables.unavailable.button${contextKey}`,
 				},
-				variables: {
+			},
+			users: {
+				settings: {
 					unavailable: {
-						title: `contextual.variables.unavailable.title${contextKey}`,
-						description: 'contextual.variables.unavailable.description',
-						action: `contextual.variables.unavailable.action${contextKey}`,
-						button: `contextual.variables.unavailable.button${contextKey}`,
+						title: `contextual.users.settings.unavailable.title${contextKey}`,
+						description: `contextual.users.settings.unavailable.description${contextKey}`,
+						button: `contextual.users.settings.unavailable.button${contextKey}`,
 					},
 				},
-				users: {
-					settings: {
-						unavailable: {
-							title: `contextual.users.settings.unavailable.title${contextKey}`,
-							description: `contextual.users.settings.unavailable.description${contextKey}`,
-							button: `contextual.users.settings.unavailable.button${contextKey}`,
-						},
-					},
-				},
-			} as const;
-		},
-		getLastSelectedNode(): INodeUi | null {
-			const workflowsStore = useWorkflowsStore();
-			if (this.lastSelectedNode) {
-				return workflowsStore.getNodeByName(this.lastSelectedNode);
+			},
+		} as const;
+	});
+
+	const getLastSelectedNode = computed(() => {
+		if (lastSelectedNode.value) {
+			return workflowsStore.getNodeByName(lastSelectedNode.value);
+		}
+		return null;
+	});
+
+	const isVersionsOpen = computed(() => {
+		return modals.value[VERSIONS_MODAL_KEY].open;
+	});
+
+	const isModalOpen = computed(() => {
+		return (name: ModalKey) => modals.value[name].open;
+	});
```

Comment:
Can we extract this to a computed property `modalOpenById` instead? (see comment about computed getters from the other PR)

It allows for caching support.

```
const modalOpenById = computed(() => {
    return Object.keys(modals.value).reduce((acc, modal) => {
        acc[modal.name] = modals.value[name].open;
        return acc;
    }, {});
});
```

Reply:
Can't we just rename modals to modalsById and use it drectly?

`uiStore.modalsById[name].open`

Reply:
We can to the same with all the other computed properties bellow:

This:

```typescript

	const getModalActiveId = computed(() => {
		return (name: ModalKey) => modals.value[name].activeId;
	});
``` 

To:

```typescript
uiStore.modalsById[name].activeId

```

This:

```typescript

	const getModalMode = computed(() => {
		return (name: ModalKey) => modals.value[name].mode;
	});
``` 
 
To:

```typescript
uiStore.modalsById[name].mode
``` 

The only one that would be to verbose to do like that would be IsModalActive. For that we can maybe catch the length and only run the comparison every time.

Reply:
Good point, I agree. Let's use it directly! 

For isModalActive we can also have a computed property:

```
const isModalActiveById = computed(() => Object.keys(modalsById.value).reduce((acc, name) => {
  acc[name] = name === modalStack.value[0];
  return acc;
}, {}); 
```